presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcPageSourceFactory.java

```diff
@@ -289,4 +291,13 @@ private static void verifyFileHasColumnNames(List<String> physicalColumnNames, P
 
         return physicalNameOrdinalMap.build();
     }
+
+    private static boolean shouldUseColumnNames(Properties schema, boolean useOrcColumnNames)
+    {
+        String columnIndexAccessProperty = schema.getProperty(OrcReaderConstants.ORC_COLUMN_INDEX_ACCESS);
```

Comment:
Orc has a similar property already: `orc.force.positional.evolution` from https://github.com/apache/orc/pull/72/files. That's on the configuration though, not the table.